argo/src/Argo.hs

```diff
@@ -83,13 +93,14 @@ jsonRPCVersion = "2.0"
 -- each of which is a function from the JSON value representing its parameters
 -- to a JSON value representing its response.
 newtype Method state result
-  = Method (ReaderT (Text -> IO ()) (StateT state IO) result)
+  = Method (ReaderT (Text -> IO (), FilePath -> IO B.ByteString) (StateT state IO) result)
```

Comment:
Would it make sense to turn this tuple into a record with named fields, like:

``` haskell
data Config { logText :: Text -> IO (), readFile :: FilePath -> IO B.ByteString }
```

It seems like this could reduce potential confusion of the purpose of those things at use sites.

Reply:
Although if the public interface in terms of `getDebugLogger` and `getFileReader` doesn't expose the `ReaderT` directly, this doesn't matter all that much.

Reply:
I see the tuple as an internal implementation detail, rather than an interface. If it gets 3 things, then I'll think a record is in order :-)